### PR TITLE
Create run_e2e_test_module.py

### PR DESCRIPTION
**Pull request by ksahilCF**
Scripts for mulltiple module

### CODE REVIEW

**Comment by ksahilCF** [comment]
@itsankit-google @rmstar please review

File: src/main/scripts/run_e2e_test_module.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    os.system("export JAVA_OPTS=-Xmx16G")
+    run_shell_command(f"sandbox/{sandbox_dir}/bin/cdap sandbox start")
+    
+
+    # Build the plugin
+    os.chdir("plugin")
+
+    doc = xml.dom.minidom.parse("pom.xml")
+    modules = doc.getElementsByTagName("module")
+    arguments_list : List[str] = []
```
Comment (itsankit-google) [inline src/main/scripts/run_e2e_test_module.py]: nit: `arguments_list` -> `modules_to_build`

File: src/main/scripts/run_e2e_test_module.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    # Run e2e tests
+    print("Preparing e2e framework")
+    os.chdir("../e2e")
+    run_shell_command("mvn clean install")
+    print("Running e2e integration tests")
+    os.chdir("..")
+    arguments_list : List[str] = []
+    if len(sys.argv) >= 1:
+        arguments_list = sys.argv[1:]
+    print("arguments list: ", arguments_list)
```
Comment (itsankit-google) [inline src/main/scripts/run_e2e_test_module.py]: Remove duplicate code

File: src/main/scripts/run_e2e_test_module.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    doc = xml.dom.minidom.parse("pom.xml")
+    modules = doc.getElementsByTagName("module")
+    arguments_list : List[str] = []
+    if len(sys.argv) >= 1:
+        arguments_list = sys.argv[1:]
+    print("arguments list: ", arguments_list)
+    for moduleName in modules:
+        if moduleName.firstChild.data in arguments_list:
+            module=moduleName.firstChild.data
+            print("Building plugin")
```
Comment (itsankit-google) [inline src/main/scripts/run_e2e_test_module.py]: nit: `print(f"Building plugin {module}")`

File: src/main/scripts/run_e2e_test_module.py (hunk 1 above)
Comment (ksahilCF) [reply src/main/scripts/run_e2e_test_module.py]: Changed the variable name to modules_to_build

File: src/main/scripts/run_e2e_test_module.py (hunk 2 above)
Comment (ksahilCF) [reply src/main/scripts/run_e2e_test_module.py]: Removed duplicate code

File: src/main/scripts/run_e2e_test_module.py (hunk 3 above)
Comment (ksahilCF) [reply src/main/scripts/run_e2e_test_module.py]: Module name included

File: src/main/scripts/run_e2e_test_module.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            print("Installing plugin")
+            res=requests.post(f"http://localhost:11015/v3/namespaces/default/artifacts/{plugin_name}", headers={"Content-Type": "application/octet-stream", "Artifact-Extends": '/'.join(plugin_parents), "Artifact-Version": plugin_version}, data=data)
+            assert res.ok or print(res.text)
+            res=requests.put(f"http://localhost:11015/v3/namespaces/default/artifacts/{plugin_name}/versions/{plugin_version}/properties", json=plugin_properties)
+            assert res.ok or print(res.text)
+            os.chdir("../..")
+               
+            # Run e2e tests
+            print("Preparing e2e framework")
+            os.chdir("../e2e")
```
Comment (itsankit-google) [inline src/main/scripts/run_e2e_test_module.py:89]: nit: `os.chdir("e2e")`

Comment (itsankit-google) [inline src/main/scripts/run_e2e_test_module.py]: nit: can we combine Line 85 and Line 89?

File: src/main/scripts/run_e2e_test_module.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            run_shell_command("mvn clean install")
+            print("Running e2e integration tests")
+            os.chdir("..")
+
+            assertion_error=None
+            os.chdir(f"plugin/{module}/e2e-test")
```
Comment (itsankit-google) [inline src/main/scripts/run_e2e_test_module.py]: This should be only `os.chdir(f"../plugin/{module}")`

File: src/main/scripts/run_e2e_test_module.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            res=requests.put(f"http://localhost:11015/v3/namespaces/default/artifacts/{plugin_name}/versions/{plugin_version}/properties", json=plugin_properties)
+            assert res.ok or print(res.text)
+            os.chdir("../..")
+               
+            # Run e2e tests
+            print("Preparing e2e framework")
+            os.chdir("../e2e")
+            run_shell_command("mvn clean install")
+            print("Running e2e integration tests")
+            os.chdir("..")
```
Comment (itsankit-google) [inline src/main/scripts/run_e2e_test_module.py]: can remove this line, see next comment.

File: src/main/scripts/run_e2e_test_module.py (hunk 4 above)
Comment (itsankit-google) [reply src/main/scripts/run_e2e_test_module.py]: Something like this `os.chdir("../../../e2e")`

File: src/main/scripts/run_e2e_test_module.py [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+    # Run e2e tests
+    print("Preparing e2e framework")
+    os.chdir("../e2e")
+    run_shell_command("mvn clean install")
+    print("Running e2e integration tests")
+    os.chdir("..")
+
+    assertion_error=None
+    os.chdir(f"plugin/{module_to_build}/e2e-test")
```
Comment (itsankit-google) [inline src/main/scripts/run_e2e_test_module.py]: It should be `os.chdir(f"plugin/{module_to_build}")`

File: src/main/scripts/run_e2e_test_module.py [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    try:
+        run_shell_command("mvn clean install -P e2e-tests")
+    except AssertionError as e:
+        assertion_error=e
+        continue
+    finally:
+        os.chdir("../../..")
```
Comment (itsankit-google) [inline src/main/scripts/run_e2e_test_module.py]: similarly here, `os.chdir("../..")`

File: src/main/scripts/run_e2e_test_module.py (hunk 4 above)
Comment (ksahilCF) [reply src/main/scripts/run_e2e_test_module.py]: done

File: src/main/scripts/run_e2e_test_module.py (hunk 5 above)
Comment (ksahilCF) [reply src/main/scripts/run_e2e_test_module.py]: Done

File: src/main/scripts/run_e2e_test_module.py (hunk 6 above)
Comment (ksahilCF) [reply src/main/scripts/run_e2e_test_module.py]: Done

File: src/main/scripts/run_e2e_test_module.py (hunk 7 above)
Comment (ksahilCF) [reply src/main/scripts/run_e2e_test_module.py]: Done

File: src/main/scripts/run_e2e_test_module.py (hunk 8 above)
Comment (ksahilCF) [reply src/main/scripts/run_e2e_test_module.py]: Done

File: src/main/scripts/run_e2e_test_module.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    # Start CDAP sandbox
+    print("Downloading CDAP sandbox")
+    sandbox_url = "https://builds.cask.co/artifact/CDAP-BUT/shared/build-latestSuccessful/SDK/cdap/cdap-standalone/target/cdap-sandbox-6.8.0-SNAPSHOT.zip"
+    sandbox_dir = sandbox_url.split("/")[-1].split(".zip")[0]
+    r = requests.get(sandbox_url)
+    z = zipfile.ZipFile(io.BytesIO(r.content))
+    z.extractall("./sandbox")
+    print("Start the sandbox")
+    run_shell_command(f"chmod +x sandbox/{sandbox_dir}/bin/cdap")
+    os.system("export JAVA_OPTS=-Xmx16G")
```
Comment (itsankit-google) [inline src/main/scripts/run_e2e_test_module.py]: nit: `16G`->`24G`

Comment (itsankit-google) [inline src/main/scripts/run_e2e_test_module.py]: nit: `16G`->`24G`

**Review by itsankit-google — COMMENTED** [review]
Overall looks good, please squash commits.

File: src/main/scripts/run_e2e_test_module.py (hunk 9 above)
Comment (ksahilCF) [reply src/main/scripts/run_e2e_test_module.py]: Done

**Review by itsankit-google — APPROVED** [review]

**Pull request merged**